### ISSUE TITLE
Container image "gcr.io/datadoghq/operator:0.7.0" already present on machine

### ISSUE THREAD

**Issue by OceaneLonneux-CAL** [DataDog/helm-charts#386]
**Describe what happened:**
We are trying to deploy datadog-operator (we want to test the new automatic monitor feature) and are running into this error `Container image "gcr.io/datadoghq/operator:0.7.0" already present on machine`.

**Describe what you expected:**
I expected the deployment to go through.

**Steps to reproduce the issue:**
Charts version:
- datadog: 2.19.4,
- datadog-crds: 0.4.1,
- datadog-operator: 0.7.0. 

Our `values.yaml`:
We have added the following 
```
datadog-operator:
  nodeSelector:
    kubernetes.io/os: linux
  datadogMonitor:
    enabled: true
```

We have tried with `image.pullpolicy: Never` as well, same result

**Additional environment details (Operating System, Cloud provider, etc):**
Akure Kubernetes version `1.21.2`.

**Comment by OceaneLonneux-CAL** [comment]
UPDATE:
I found what was wrong:
You need to add the following under datadog-operator. 
```
  apiKey: XXX
  appKey: XXX
```

Should have read the docs more carefully, will close this.

**Comment by OceaneLonneux-CAL** [comment]
Re-opening because the API addition did help, but we're still seeing the "Container image "gcr.io/datadoghq/operator:0.7.0" already present on machine". 

Tried again with the `PullPolicy: Never` but did not help. 

It goes `running` then crash loop continuously (which is why I closed this issue as I saw it was green)

**Comment by clamoriniere** [comment]
Hello @OceaneLonneux-CAL 

The info `Container image "gcr.io/datadoghq/operator:0.7.0" already present on machine` is harmless. it just means that the container runtime don't need to pull again the image, because the exact image tag is already. 
Even if you set `PullPolicy: Never` you will see this message. So I recommend you to set `PullPolicy: IfNotPresent` because if the operator pod is reschedule on another Node, it will not be able to start.

Now let's try to understand why the Operator pod is in "crash loop"
Could you provide the output of the commands:
* `kubectl describe pod <operator-pod-name>` => it will help to know if the container restart due to an error, or a Out Of Memory issue.
* `kubectl logs -p  <operator-pod-name>` => it will help to get the error context. (maybe a stack-trace)

**Comment by OceaneLonneux-CAL** [comment]
Hi @clamoriniere, thanks for answering!

I can actually see something up in the logs:
```
"logger":"setup","msg":"starting manager"}
"logger":"klog","msg":"attempting to acquire leader lease infrastructure/datadog-operator-lock...\n"}
"logger":"controller-runtime.manager","msg":"starting metrics server","path":"/metrics"}
"logger":"klog","msg":"successfully acquired lease infrastructure/datadog-operator-lock\n"}
"logger":"controller-runtime.manager.controller.datadogmonitor","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogMonitor","source":"kind source: /, Kind="}
"logger":"controller-runtime.manager.controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: /, Kind="}
"logger":"controller-runtime.manager.controller.datadogmonitor","msg":"Starting Controller","reconciler group":"datadoghq.com","reconciler kind":"DatadogMonitor"}
"logger":"controller-runtime.source","msg":"if kind is a CRD, it should be installed before calling Start","kind":"DatadogAgent.datadoghq.com","error":"no matches for kind \"DatadogAgent\" in version \"datadoghq.com/v1alpha1\""}
"logger":"controller-runtime.manager.controller.datadogmonitor","msg":"Could not wait for Cache to sync","reconciler group":"datadoghq.com","reconciler kind":"DatadogMonitor","error":"failed to wait for datadogmonitor caches to sync: cache did not sync"}
"logger":"controller-runtime.manager","msg":"error received after stop sequence was engaged","error":"failed to wait for datadogmonitor caches to sync: cache did not sync"}
"logger":"setup","msg":"Problem running manager","error":"no matches for kind \"DatadogAgent\" in version \"datadoghq.com/v1alpha1\""}

```

**Comment by clamoriniere** [comment]
hi @OceaneLonneux-CAL 

if seems that the `DatadogAgent` CRD is not install on you cluster, but it should be done automatically when you install the operator. 

could you run: `kubectl get crd datadogagents.datadoghq.com`  and `kubectl get crd datadogmetrics.datadoghq.com`

if they are not installed, could you try to reinstall the `datadog-operator` chart. (latest chart version 0.7.0).

**Comment by OceaneLonneux-CAL** [comment]
I can indeed see that the customresources are missing.

For context, here is how we operate:
We have community helm charts, then we have a wrapper chart that will "unite" our use of Datadog. So that wrapper would have in its values datadog, datadog-operator, datadog-crds. This allows us to not change directly the community code and just pull the code and pin a version easily.

I have re-installed the 0.7.0 datadog-operator. Still missing the CR when I use `helm template .`. There is something I noticed:
- I pulled down in my files version 0.7.0 of `datadog-operator`,
- in our wrapper charts, I filled in the `values.yaml` and added `datadog-operator` version 0.7.0 to our `requirements.yaml`.

When I `helm dep update`, I did not see the dependency of datadog-crds getting downloaded in our wrapper-charts (the dependency is in datadog-operator). So I added to our  `requirements.yaml` datadog-crds version 0.4.1 and the CR are still missing.

**Comment by clamoriniere** [comment]
Hello I think I know the issue, 

We are almost here :)

The fact that the DatadogMonitor status is empty give us the information that the Operator didn't process it.
```
NAME                   ID    MONITOR STATE   LAST TRANSITION   LAST SYNC   SYNC STATUS   AGE
datadog-monitor-test                                                                     4m5s
```

You need to add your `values.yaml` file

```yaml
datadogMonitor:
  enabled: true
```

more documentation [here](https://github.com/DataDog/datadog-operator/blob/main/docs/datadog_monitor.md#adding-a-datadogmonitor) . As explained in `1. ii` only 3  parameters is requires: `datadogMonitor.enabled:true`, `apiKey`, `appKey`

```yaml
apiKey:<DATADOG_API_KEY> 
appKey:<DATADOG_APP_KEY>

datadogMonitor:
  enabled: true
```

Also, I think that you can remove the following line in your `values.yaml` since it correspond to the default values.

```yaml
datadog-crds:
  crds:
    datadogMonitors: true
    datadogAgents: true
```

Please let us know if you have the Operator running properly with DatadogMonitor

**Comment by OceaneLonneux-CAL** [comment]
Thanks again for your help, I do feel like we are getting closer!

I actually already had the config you mentioned:
```
datadog-operator:
  nodeSelector:
    kubernetes.io/os: linux
  apiKey: xxx
  appKey: xxx
  datadogMonitor:
    enabled: true

datadog-crds:
  crds:
    datadogMonitors: true
    datadogMetrics: true
    datadogAgents: true
```
This is how this is deployed and giving the empty resource result.

**Comment by celenechang** [comment]
Hi @OceaneLonneux-CAL , I just did a test with the provided values.yaml file; do you mind trying a `helm install/upgrade` with this?

```
nodeSelector:
  kubernetes.io/os: linux
apiKey: XXX
appKey: XXX
datadogMonitor:
  enabled: true

datadog-crds:
  crds:
    datadogMonitors: true
    datadogMetrics: true
    datadogAgents: true
```

With this configuration, along with the DatadogMonitor manifest you provided above, I was able to get it to work.

Let us know how it goes!

**Comment by OceaneLonneux-CAL** [comment]
Hi @celenechang, thank you for answering.

We had the same configuration you mentioned. I actually just found out what was wrong:
In the datadog-operator pod, I noticed this line in the logs:
`Manager will be watching namespace","namespace":"infrastructure"`. The test.yaml I was using from your example is using default namespace, and we deploy the `datadog-operator` in a specific namespace. So specifying the namespace in the yaml fixed it, and datadog-operator "saw" the new `DatadogMonitor`

Is there any way to make this more obvious in the documentation or working in the same namespaces is something that should always be done?
